report/sections/Evaluation/Benchmark.tex

```diff
@@ -0,0 +1,196 @@
+However, the internal implementation of the client varies, as it needs to pull the results of the Test Runner after making an initial request.
+Therefore, an additional parameter for the benchmark is introduced to examine how often a client should try to pull for a result.
+Best case pull time needs to be established to compare it with the baseline component. 
+However, due to the nature of remote communication using the internet it may be necessary to establish which polling times to avoid rather than which gives the best average response times, since the benchmarks does not consider the physical distance between client and server.
+\begin{lstlisting}[language=CSharp, escapechar=~, caption={C\# code showing the implementation of the benchmark for the backend with the Queue System }, label={lst:queueSystemBench}]
+// Attributes excluded
+public class TicketedCodeRunnerBenchmark
+{
+    [Params(0.5, 1.0, 2.0, 3.0)] 
+    public double PollTime { get; set; }
+    [Params(10, 20, 50, 100)] 
+    public int NumberOfRequests { get; set; }
+    [ParamsSource(nameof(CodeSubmissions))] 
+    public CodeSubmission CodeSubmission { get; set; }
+    public static IEnumerable<CodeSubmission> CodeSubmissions => CodeLoader.Load();
+
+    [Benchmark]
+    public void PostAndWaitForAllResultsFetched()
+    {
+      TimeSpan timeBetweenPulls = TimeSpan.FromSeconds(PollTime);
+      IEnumerable<Task> clientActions = TaskBuilder.BuildClientTaskList<CodeRunnerQueueClient>(NumberOfConcurrentRequests, 
+        client => client.PostAndGetHaskellResultTask(CodeSubmission.code, CodeSubmission.test, timeBetweenPulls));
+
+      Task.WhenAll(clientActions).Wait();
+    }
+}
+\end{lstlisting}
+
+\subsection{Results}
+In this section, results of the combined stress tests and benchmarks are presented.
+First, the configuration of the stress tests are described followed by a presentation of the results of the initial Test Runner benchmark results. These results will be used to compare the Test Runner using the Queue System with the baseline.  
+
+\subsection{Configuration}
+The baseline uses a single thread to test the submitted Haskell code, thus no parameterization for this version of the Test Runner is necessary.
+Benchmarks for the Test Runner implementing the Queue System is configured such that the number of threads and the queue size varies by five. 
+All stress tests and benchmarks are run on computers with an AMD EPYC Processor (with IBPB), four CPUs, four logical cores, and four physical cores.
+The computers used for the benchmarks use UBUNTU 20.04 as the operating system and have 32GB of RAM.
+All stress tests begins with a warm-up phase, where BenchmarkDotNet sends ten consecutive submissions to the Test Runner. 
+Then, for each combination of parameters (number of requests, code submission, and polling times the stress test is timed and repeated 40 times.
```

Comment:
Missing bracket somewhere - also why are these brackets even here in the first place?